src/implementations/wrap/wrap.ts

```diff
@@ -193,19 +219,40 @@ export const wrap = async ({
 
   // Phase 1: For each document, read content, digest and write to file
   const schema = await loadSchema(schemaPath);
-  const individualDocumentHashes = await digestDocument(inputPath, intermediateDir, version, unwrap, schema);
+  const individualDocumentHashes = await digestDocument(
+    inputPath,
+    // if we dont batch document we can directly write to the destination folder.
+    batched ? intermediateDir : outputPath,
+    version,
+    unwrap,
+    batched ? Output.Directory : outputPathType,
+    schema
+  );
 
   if (!individualDocumentHashes || individualDocumentHashes.length === 0)
     throw new Error(`No documents found in ${inputPath}`);
 
-  // Phase 2: Efficient merkling to build hashmap
-  const hashMap = merkleHashmap(individualDocumentHashes);
+  let merkleRoot: string | undefined;
+  if (batched) {
```

Comment:
Cant we return if its not batched?

Reply:
Return what?

Reply:
I will rephrase what do you want to return

Reply:
will leave like this